--- script_03_tidied.pl	2020-12-08 22:15:11.467291509 +0100
+++ script_04_solved.pl	2020-12-08 22:27:33.604678583 +0100
@@ -39,6 +39,7 @@
 
 sub _s {
     ( $b, $bc, $x, $y ) = @_;
+    print { *STDERR } $bc unless '#' eq $bc;
     for $yy ( 0 .. 2 ) {
         for $xx ( 0 .. 5 ) {
             substr( $f[ $yy + $y ], ( $xx + $x ), 1 ) = $bc
@@ -101,7 +102,7 @@
 n();
 while (1) {
     $k = k();
-    last if ( $k =~ /q/ );
+    #last if ( $k =~ /q/ );
     $k  = substr( $k, 2, 1 );
     $dx = ( $k eq 'C' ) - ( $k eq 'D' );
     $bx += $dx unless ( cb( $__, $bx + $dx, $by ) );
@@ -113,7 +114,7 @@
     bl( $__, $_b, 101 + $bi, $bx, $by );
     select( undef, undef, undef, 0.1 );
     if ( cb( $__, $bx, ++$by ) ) {
-        last if ( $by < 2 );
+        #last if ( $by < 2 );
         _s( $__, $_b, $bx, $by - 1 );
         n();
     }
